### PR TITLE
Fixed IsFloatPositiveZero from returning 'true' on non-constant double operands

### PR DESCRIPTION
**Pull request by TIHan**
This is a quick bug fix - the logic for `IsFloatPositiveZero` is incorrect as it would return 'true' for operands that are not non-constant doubles.

### CODE REVIEW

File: src/coreclr/jit/gentree.h [hunk 1]
```diff
@@ -7661,7 +7661,12 @@ inline bool GenTree::IsSIMDZero() const
 //
 inline bool GenTree::IsFloatPositiveZero() const
 {
-    return !(IsCnsNonZeroFltOrDbl());
+    if (OperGet() == GT_CNS_DBL)
+    {
+        return !(IsCnsNonZeroFltOrDbl());
+    }
+
+    return false;
```
Comment (EgorBo) [inline src/coreclr/jit/gentree.h:7669]:
```suggestion
    return OperIs(GT_CNS_DBL) && IsCnsNonZeroFltOrDbl();
```

**Review by tannergooding — APPROVED** [review]

File: src/coreclr/jit/gentree.h [hunk 2]
```diff
@@ -7661,7 +7661,7 @@ inline bool GenTree::IsSIMDZero() const
 //
 inline bool GenTree::IsFloatPositiveZero() const
 {
-    return !(IsCnsNonZeroFltOrDbl());
+    return IsCnsFltOrDbl() && !IsCnsNonZeroFltOrDbl();
```
Comment (echesakovMSFT) [inline src/coreclr/jit/gentree.h:7664]: I wonder if we just should explicitly implement the logic?
```c++
    if (IsCnsFltOrDbl())
    {
        double constValue = AsDblCon()->gtDconVal;
        return *(__int64*)&constValue == 0;
    }
```

For me personally, a conjunction of *"is a floating point constant?"* and *"is not a non-zero floating point constant"* is tricky to parse mentally.

Comment (echesakovMSFT) [reply src/coreclr/jit/gentree.h:7664]: If you agree you can follow up after the snap - don't want to block the PR on that.

**Review by echesakov — APPROVED** [review]

File: src/coreclr/jit/gentree.h (hunk 2 above)
Comment (TIHan) [reply src/coreclr/jit/gentree.h:7664]: Yea, we can follow up and I agree that the current logic is hard to parse mentally which is probably why there was a bug :)
